src-java/kilda-model/src/main/java/org/openkilda/model/FlowMirror.java

```diff
@@ -113,6 +116,25 @@ public boolean isOneSwitchMirror() {
         return getMirrorSwitchId().equals(getEgressSwitchId());
     }
 
+    /**
+     * Return opposite pathId to passed pathId.
+     */
+    public Optional<PathId> getOppositePathId(PathId pathId) {
+        Optional<Long> requestedPathCookie = getPath(pathId)
+                .map(FlowMirrorPath::getCookie)
+                .map(FlowSegmentCookie::getFlowEffectiveId);
+        if (requestedPathCookie.isPresent()) {
+            return getMirrorPaths().stream()
+                    .filter(path -> !path.getMirrorPathId().equals(pathId))
+                    .filter(path -> path.getCookie().getFlowEffectiveId() == requestedPathCookie.get())
+                    .findAny()
+                    .map(FlowMirrorPath::getMirrorPathId);
+        } else {
+            throw new IllegalArgumentException(format("Flow mirror %s does not have mirror path %s",
```

Comment:
I am confused here. The else branch says there is no a single mirror path. If it is not the case, then there is at least one mirror path, then how an empty Optional could happen?
The exception message probably should be about path cookies and not about mirror path?

Reply:
not really. 
```
        Optional<Long> requestedPathCookie = getPath(pathId)
                .map(FlowMirrorPath::getCookie)
                .map(FlowSegmentCookie::getFlowEffectiveId);
```
`requestedPathCookie` will be empty only if `getPath(pathId)` will return empty optional. That means there is no path with pathId `pathId`

Reply:
Right, but if there is no path with this `pathId ` as you described, then it is the situation described in the exception, because it is true that the `Flow mirror %s does not have the mirror path %s`, but the exception is not thrown.